### DISCUSSION TITLE
feature request: journal templates

### DISCUSSION THREAD

**Discussion by sanzoghenzo** [GitJournal/GitJournal#796] [Feature Requests]
I'm a big fan of the 5 minutes journal, and I would like to have a base template to start with when I create a new note.

As a workaround to that, I tried to use a couple of text expanding app, but it's not so great because the instant text expand feature works only in native and standard text input component.

In my mind, this could be done by adding a "templates" item in the settings to create as many templates as one wants, and if there's at least one of them, then the "+" button should provide a "blank entry" and "from template".

**Comment by vHanda** [comment]
Hey Andrea

This sounds like a great idea. I quite like how Evernote provides you an
option to easily choose a template. Maybe something similar could work for
GitJournal.

**Comment by devoldski** [comment]
Templates could in my opinion be the ability to have preconfigured yaml front mappers. that could be inserted

**Comment by vHanda** [comment]
Hey devoldski. I quite like the idea of extending template to YAML metadata. I'll keep it in mind when I start implementing it. For now - Folder support is a higher priority.

**Comment by devoldski** [comment]
thank you for this great app and support. 
Regarding  theming/template you could add meta info such as "Template: journal" in YAML  or any other template name that makes it possible to extend for different use cases.

**Comment by taro-yjp** [comment]
Templates sounds very good.
for example, say when I want to write about a movie just saw, 

```yaml 
---
movie title:
theater:
my rating:
---
```
Just as current "title" section, users can add what they want.
The first element will be "title" of the note and others will be showed as bullet list like

```yaml
---
movie title: Joker
theater: Some movie theater
my rating: 10/10
---
```
showed as:
### Joker
- Some movie theater
- 10/10

Also, current version of Git journal have an ability to handle folders, I'd like to each folder would have it's own template(s).

I like the simplicity of the application, so when just tapping "+" button creates default blank note, hold "+" button to select template dialog (or something) popped up would be nice.

**Comment by vHanda** [comment]
It would be easier if this was implemented in the same way as Obsidian's templates - https://publish.obsidian.md/help/Plugins/Templates

**Comment by mc0e** [comment]
Lack of templates is the primary reason why I haven't yet adopted git-journal.  I come back to looking at git-journal over and over though, because in so many ways it is the way I'd like to go for keeping my notes.

**Reply by ThomUK** [reply]
What alternative have you found? I'm in a similar position.

**Reply by codey-badger** [reply]
@ThomUK @mc0e what did you end up using? I'm in the same boat.
